### PR TITLE
update setup.rst

### LINKED ISSUE #16178: mention --production arg for yarn in doc/setup.rst
from:  `doc/setup.rst`

> If you do not intend to develop, you can skip the installation of developer tools by invoking:
composer update --no-dev
Finally, you'll need to use Yarn to install some JavaScript dependencies:
yarn install

There is mention of `--no-dev` for composer, but no mention of argument `--production` for yarn to avoid installing devDependencies if the user not intend to develop/test the software.

### PR DESCRIPTION
**Pull request by jabbarn**
### Description

Please describe your pull request.
 Adding the --production flag in the setup.rst to make the docs clearer.
Fixes #16178

Before submitting pull request, please review the following checklist:

- [X] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [X] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [X] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [X] Every commit has a descriptive commit message.
- [X] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [X] Any new functionality is covered by tests.

### CODE REVIEW

**Comment by williamdes** [comment]
@DaemonicNaymon Could you do the following commands:
```
git reset --hard origin/QA_5_0
git cherry-pick fb43c2a5b8329ff408782094028ecc76dae94dd4
git push --force
```

To comply with "Your pull request should target the QA_5_0 branch." https://github.com/phpmyadmin/phpmyadmin/issues/16178#issuecomment-644883823

Thank you for the pull-request :)

File: doc/setup.rst [hunk 1]
```diff
@@ -122,8 +122,9 @@ Finally, you'll need to use `Yarn`_ to install some JavaScript dependencies:
 
 .. code-block:: sh
 
-    yarn install
+    yarn install --production
 
+The --production flag is used to ignore the develop dependencies in `Yarn`_.
```
Comment (ibennetch) [inline doc/setup.rst:127]: In reflecting on this, I don't think we need the explanation at all here; just adding `--production` to the yarn command should be sufficient. That's what we do for composer, without any further explanation.

@williamdes what do you think?

Comment (williamdes) [reply doc/setup.rst:127]: Well the `--only=prod` from npm was more explicit, I would agree to keep the explanation to add more clarity about this flag
I know a lot of people that would only use `yarn` command ignoring the `--production` flag by simple laysiness.
Anyway, keep it or not I agree :)

Comment (DaemonicNaymon) [reply doc/setup.rst:127]: Well a bit more clarity won't hurt since it only adds more information. So I guess I'll leave it as it is. :)

Comment (mauriciofauth) [reply doc/setup.rst:127]: I agree with Isaac.

**Comment by DaemonicNaymon** [comment]
it seems that a check is failing, can I do anything about it so i can close this 😅?

**Comment by williamdes** [comment]
> it seems that a check is failing, can I do anything about it so i can close this ?

Do not worry `codecov/project` can be ignored :)

**Comment by williamdes** [comment]
@DaemonicNaymon can you remove the description line so I can merge your work ?

**Comment by DaemonicNaymon** [comment]
oh yes of course, give me a sec, I have to push it to QA_5_0 right?

**Review by williamdes — APPROVED** [review]

**Comment by DaemonicNaymon** [comment]
Is it safe to close this PR? 😅

**Comment by williamdes** [comment]
Hi @DaemonicNaymon !
Thank you for your contribution to our documentation, if you want to contribute more you should look at https://github.com/phpmyadmin/phpmyadmin/pull/15942/files and maybe consider deleting and creating the fork a new time (or hard reset all main branches to the main repo)

Because you used QA_5_0 to do this pull-request and that screwed the history :)

**Comment by DaemonicNaymon** [comment]
AH I'm so sorry @williamdes , this is my first time contributing to an open source project and thought I could manage a simple issue like that. If I have been an inconvenience, I would like to apologize to you.

**Comment by williamdes** [comment]
> AH I'm so sorry @williamdes , this is my first time contributing to an open source project and thought I could manage a simple issue like that. If I have been an inconvenience, I would like to apologize to you.

No worries, I can understand and in the end it is more for you and the maintenance of further contributions
We prefer that contributions do not use the branch names that we use, to allow easy rebasing or changes of the branch

**Pull request merged**